app/models/requests/recap_item_availability.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+module Requests
+  # This class is responsible for finding availability data from SCSB for items in ReCAP (whether owned by PUL or a partner)
+  class RecapItemAvailability
+    include Scsb
+
+    def initialize(id:, scsb_location:)
+      @id = id
+      @scsb_location = scsb_location
+    end
+
+    # Returns an array of item hashes
+    def items_with_availability(items:)
+      items.map { |item| ItemWithAvailability.new(availability_data:, item:).call }
+    end
+
+    # This inner class is responsible for adding availability data that has already been fetch from SCSB to an item hash
+    class ItemWithAvailability
+      def initialize(availability_data:, item:)
+        @availability_data = availability_data
+        @item = item
+      end
+
+      def call
+        # Not sure why we set the string and symbol versions to different values...
```

Comment:
I think because we need to do some work in bibdata first and bring the status label that we want.